### PR TITLE
Adding functional tests to MetaPackages

### PR DESCRIPTION
**Pull request by JunTaoLuo**
This is following up to #24  For the new static WebHost APIs

### CODE REVIEW

File: test/TestSites/CreateDefaultBuilderApp/Program.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                return $"appsettings.json not loaded into Configuration.";
+            }
+
+            // Verify appsettings.environment.json loaded
+            if (!string.Equals("devSettingsValue", context.Configuration["devSettingsKey"], StringComparison.Ordinal))
+            {
+                return $"appsettings.{context.HostingEnvironment.EnvironmentName}.json not loaded into Configuration.";
+            }
+
+            // Verify UserSecrets loaded
```
Comment (JunTaoLuo) [inline test/TestSites/CreateDefaultBuilderApp/Program.cs]: This gets really tricky to test.

File: test/TestSites/CreateDefaultBuilderApp/Program.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                })
+                .Configure(app =>
+                {
+                    app.Map("/error", subApp =>
+                    {
+                        subApp.Run(context => throw new Exception());
+                    });
+                    app.Run(async context =>
+                    {
+                        await context.Response.WriteAsync(responseMessage);
```
Comment (Tratcher) [inline test/TestSites/CreateDefaultBuilderApp/Program.cs]: unneeded await

File: test/TestSites/StartRequestDelegateApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace StartRequestDelegateApp
```
Comment (Tratcher) [inline test/TestSites/StartRequestDelegateApp/Program.cs]: Need to use dynamic ports, not defaults

File: test/TestSites/StartRequestDelegateUrlApp/Program.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace StartRequestDelegateUrlApp
+{
+    public class Program
+    {
+        static void Main(string[] args)
+        {
+            var messageSent = new ManualResetEventSlim(false);
+
+            using (WebHost.Start("http://localhost:5001", async context =>
```
Comment (Tratcher) [inline test/TestSites/StartRequestDelegateUrlApp/Program.cs]: Dynamic port from env.

File: test/TestSites/StartWithIApplicationBuilderUrlApp/Program.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                {
+                    var env = context.RequestServices.GetRequiredService<IHostingEnvironment>();
+                    await context.Response.WriteAsync(env.ApplicationName);
+                    messageSent.Set();
+                });
+            }))
+            {
+                // Need these for test deployer to consider host deployment successful
+                // The address written here is used by the client to send requests
+                Console.WriteLine("Now listening on: http://localhost:5001");
```
Comment (Tratcher) [inline test/TestSites/StartWithIApplicationBuilderUrlApp/Program.cs]: Echo the real address

**Comment by JunTaoLuo** [comment]
Adding test for #24 which is in preview1. This should also go into the 2.0.0-preview1 branch cc @muratg @Eilon

File: test/TestSites/CreateDefaultBuilderApp/Program.cs (hunk 1 above)
Comment (muratg) [inline test/TestSites/CreateDefaultBuilderApp/Program.cs]: Is this a "TODO"?

**Pull request merged**